onnxruntime/core/providers/webnn/builders/impl/gemm_op_builder.cc

```diff
@@ -42,13 +42,54 @@ Status GemmOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder, const N
     if (!GetShape(*input_defs[a_idx], a_shape, logger)) {
       return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT, "Can not get shape of A.");
     }
+    std::vector<int64_t> b_shape;
+    if (!GetShape(*input_defs[b_idx], b_shape, logger)) {
+      return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT, "Can not get shape of B.");
+    }
+    // If the first argument is 1-D, it is promoted to a matrix by prepending a 1 to its dimensions.
+    bool prepend_flag = false;
+    if (a_shape.size() == 1) {
+      prepend_flag = true;
+      a_shape.insert(a_shape.begin(), 1);
+      a = model_builder.GetBuilder().call<emscripten::val>("reshape", a,
+                                                           emscripten::val::array(GetVecUint32FromVecInt64(a_shape)));
+    }
+    // If the second argument is 1-D, it is promoted to a matrix by appending a 1 to its dimensions.
+    bool append_flag = false;
+    if (b_shape.size() == 1) {
+      append_flag = true;
+      b_shape.push_back(1);
+      b = model_builder.GetBuilder().call<emscripten::val>("reshape", b,
+                                                           emscripten::val::array(GetVecUint32FromVecInt64(b_shape)));
+    }
     // The inputs of MatMul must be at least 3D for WebNN CPU backend. Use GEMM for 2D case.
     // TODO: Remove this workaround when it is fixed in Chromium.
     if (model_builder.GetWebnnDeviceType() == WebnnDeviceType::CPU && a_shape.size() == 2) {
       output = model_builder.GetBuilder().call<emscripten::val>("gemm", a, b);
     } else {
       output = model_builder.GetBuilder().call<emscripten::val>("matmul", a, b);
     }
+    // If the inputs are both 1D， reduce the output to a scalar.
+    if (prepend_flag && append_flag) {
+      output = model_builder.GetBuilder().call<emscripten::val>("reduceMean", output);
```

Comment:
It is a 1d*1d tensor, so both ops are valid.

Reply:
> both ops are valid

You are correct (because x/1 = x), but it's wasteful to divide /1 unnecessarily, and semantically it doesn't make sense because no reduction occurs if the input is 1x1 (as there was 1 element before and 1 element after). Isn't this really just a `reshape` though?

Reply:
Yes, reshape is the better. I was concerned about the scalar support of WebNN reshape. I have just tested with this case, and it works.